plugins/dell-dock/fu-dell-dock-i2c-ec.c

```diff
@@ -455,22 +485,28 @@ fu_dell_dock_ec_get_dock_info (FuDevice *device,
 
 	/* Thunderbolt SKU takes a little longer */
 	if (self->data->module_type == MODULE_TYPE_130_TBT ||
-	    self->data->module_type == MODULE_TYPE_45_TBT) {
+	    self->data->module_type == MODULE_TYPE_45_TBT  ||
+	    self->data->module_type == MODULE_TYPE_130_USB4) {
 		guint64 tmp = fu_device_get_install_duration (device);
 		fu_device_set_install_duration (device, tmp + 20);
 	}
 
 	/* minimum EC version this code will support */
+	if (self->data->module_type == MODULE_TYPE_130_USB4)
+		opt_ec_minimum_version = self->ec_minimum_version_usb4;
```

Comment:
I'm actually not convinced that all the code for the minimum EC version for USB4 module makes sense to check here.  If you have a version without USB4 support doesn't that mean it won't know about USB4 enums/types/etc?  

Where it _might_ be useful to check is during the write function.

If the user is trying to install an "older" EC image and you have USB4 module then it shouldn't be allowed to install. However I think that idea falls apart if the EC image used doesn't start at _least_ with the latest image that is available for WD19 today.  Development versions of 00.xx.yy.zz it breaks.

So in summary; I think you should drop this idea for now, and consider to add it back in for safety later.

Reply:
One more thought coming to mind though - if development ECs are around less than 01.00.00.00 for USB4 module, then maybe the minimum version check should be **removed all together** when the module is USB4..

Reply:
> If you have a version without USB4 support doesn't that mean it won't know about USB4 enums/types/etc? Where it might be useful to check is during the write function.

I think `dock info` in general is well supported. The check for `ec_minimum_version` is more useful in `write` func. to prevent user accidentally downgrade to USB4's development version via `--force`.

> if development ECs are around less than 01.00.00.00 for USB4 module, then maybe the minimum version check should be removed all together when the module is USB4..

At this time I think `ec_minimum_version` in `write` func. is useful for `none-USB4` only. Depends on the need of USB4 hardware if there's a denial to downgrade to a particular version then we can add it back.

Reply:
> I think dock info in general is well supported. The check for ec_minimum_version is more useful in write func. to prevent user accidentally downgrade to USB4's development version via --force.

So perhaps you can switch the logic around to move the check for minimum version to write func and set up logic something like this:
1. Check for force and ignore all checks if so.
2. If not force: module type is USB4, ignore checks if so.
3. If not force and not usb4 module check If target version is less than minimum version AND current version is greater than minimum version
 
And then later when USB4 EC is stable and closer to launch this logic an be adjusted again.

Reply:
Done, thanks for the flow suggestion.